src/libraries/System.Private.CoreLib/src/System/SearchValues/Strings/SingleStringSearchValuesThreeChars.cs

```diff
@@ -325,7 +327,8 @@ private bool TryMatch(ref char searchSpaceStart, int searchSpaceLength, ref char
 
                 ValidateReadPosition(ref searchSpaceStart, searchSpaceLength, ref matchRef, _value.Length);
 
-                if (TCaseSensitivity.Equals(ref matchRef, _value))
+                if ((typeof(TCaseSensitivity) == typeof(CaseSensitive) && !TValueLength.AtLeast4Chars) ||
```

Comment:
I'm missing why this can be special-cased to avoid a comparison... is it because we know when this method is called that at least 3 characters have matched, and thus if we know the value isn't at least 4 characters, we know exactly 3 have matched?  Why the CaseSensitive limitation in that case?  A comment would be helpful.

Reply:
Yes, if the value length is 2 or 3, then the 3 anchor characters we've just checked already represent the whole value.

With case-insensitive comparisons, we just applied a mask to the input (not an actual full case conversion). That means anchor characters are _likely_ to have matched, but we don't know for sure.